### PR TITLE
New Race Map with Bots

### LINKED ISSUE #9: Speed boost
4.2 As a user, I want to be able to boost the speed on the car, so that I can gain a lead on my opponent.

Acceptance criteria: The smart-car gains a speedboost when driving over a marked area

FR: 1.6

### LINKED ISSUE #10: Bot driver
4.3 As a user, I want to be able to race against a bot, so that I can race without another player.

### LINKED ISSUE #23: Customize Environment
Customize environment with GODOT to fit our vision for the smartcar.

### LINKED ISSUE #73: racing environment Mini-map
4.8 As a user I would like to be able to see the position of the other racers so that i can tell where I am relative to them during the race.
Acceptance criteria: The cars spawned are all visible to user on track and mini-map.

FR 1.22

### LINKED ISSUE #75: Make cars more visible on mini-map
Improvement to user story 4.8 "As a user I would like to be able to see the position of the other racers so that i can tell where I am relative to them during the race." (issue #73) to make the cars on the map more visible to the player

### PR DESCRIPTION
**Pull request by drakeaxelrod**
This new smce mod includes a new race track with cars that drive
themselves. booster plates to increase the speed of smce during the
race. a minimap to keep track of the posisition of everyone. a clock
that keeps track of the time of the race and each cars finish time.

### CODE REVIEW

**Comment by klarasve** [comment]
Fully functional and works, just need to update one line in arduino file to make mqtt work and show up on website:
`vroom.ino line 79: mqtt.onMessage(+[](String& topic, String& message) `
(by adding + and &)

**Review by klarasve — CHANGES_REQUESTED** [review]
Fully functional and works, just need to update one line in arduino file to make mqtt work and show up on website:
vroom.ino line 79: `mqtt.onMessage(+[](String& topic, String& message)`
(by adding + and &)
Also looks really, really good and introduces a lot more functionalities such as minimap and bot cars!

File: godot-resources/mods/RaceAgainstMod/src/environments/raceagainst/replacements/smoke.gd [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+# Called when the node enters the scene tree for the first time.
+func _ready():
+	pass # Replace with function body.
+
+
+# Called every frame. 'delta' is the elapsed time since the previous frame.
+func _process(delta):
+
+	if ray.is_colliding() or ray2.is_colliding() or ray3.is_colliding() :
+		$smoke/Smoke.emitting = true
```
Comment (VernitaJ) [inline godot-resources/mods/RaceAgainstMod/src/environments/raceagainst/replacements/smoke.gd:18]: I really like this added feature. Great for the gaming environment

**Review by VernitaJ — APPROVED** [review]
This implementation is working well. The countdown and placement display are great additions for the gaming aspect. Speed boost is great, it's increased the fun in the application massively!
Some small additions to be added to this part of the project including difficulty variations, but that will be separate to this PR. Great job!

**Review by klarasve — APPROVED** [review]
Fully functional and works.

**Review by Axel-J-L — APPROVED** [review]
Great job with this Drake, appreciate working with you on this!

**Pull request merged**